lib/libc/minimal/include/errno.h

```diff
@@ -28,14 +28,15 @@
  * @{
  */
 
+/* errno_private.h depends on errno being defined first. */
+#define errno (*z_errno())
+
```

Comment:
This doesn't make sense to me -- the only place errno_private.h uses errno directly is in defining `z_errno`, so this define would result in a circular definition?

Reply:
Let me see if I can retrace this (it took me a long time to get Zephyr to build with Bazel so this change was pretty old.

- `lib/libc/minimal/include/errno.h` includes `zephyr/sys/errno_private.h`
- `zephyr/sys/errno_private.h` with `CONFIG_LIBC_ERRNO` includes `errno.h` which is a no-op since we have already `#define ZEPHYR_LIB_LIBC_MINIMAL_INCLUDE_ERRNO_H_`
- Then `zephyr/sys/errno_private.h` defines `z_errno()` which returns `&errno` but `errno` is only defined after `zephyr/sys/errno_private.h` is included.

Reply:
I'm curious to know why Zephyr hasn't encountered this error before.

Reply:
But we don't want `errno` defined as `(*z_errno())` when we define `z_errno` as that's just an infinite loop. If you want to use CONFIG_LIBC_ERRNO, you need the minimal C library to have its own definition of errno, presumably as a global variable. Probably something like this:

```
#ifdef CONFIG_LIBC_ERRNO
int errno;
#endif
#include <zephyr/sys/errno_private.h>
#define errno (*z_errno())
```

Reply:
Ok, let me rebase and try without this commit. I'll report back tonight.

Reply:
You're right, this must have been an old artifact from when I was trying to split up the build. Sorry.